From 606623bf8e9c1cda176fc0298228a1ebd9b57409 Mon Sep 17 00:00:00 2001
From: Sean Du <do4suki@gmail.com>
Date: Thu, 10 Sep 2026 00:38:18 +0800
Subject: [PATCH 8/8] Fix VerifyError in FilterMaskView Record class

Replace broken invoke-custom calls with manual implementations for
equals(), hashCode(), and toString() methods to fix bytecode verification
error caused by incorrect outsSize in the Record class.

Signed-off-by: Sean Du <do4suki@gmail.com>
---
 .../camera/feature/filter/view/f0$b$a.smali   | 314 +++++++++++++++++-
 1 file changed, 302 insertions(+), 12 deletions(-)

diff --git a/smali_classes31/com/oplus/camera/feature/filter/view/f0$b$a.smali b/smali_classes31/com/oplus/camera/feature/filter/view/f0$b$a.smali
index b729f53b..21c903aa 100644
--- a/smali_classes31/com/oplus/camera/feature/filter/view/f0$b$a.smali
+++ b/smali_classes31/com/oplus/camera/feature/filter/view/f0$b$a.smali
@@ -162,46 +162,336 @@
 .end method
 
 .method public final equals(Ljava/lang/Object;)Z
-    .locals 0
+    .locals 3
 
     .line 1
-    invoke-custom {p0, p1}, call_site_0("equals", (Lcom/oplus/camera/feature/filter/view/f0$b$a;Ljava/lang/Object;)Z, Lcom/oplus/camera/feature/filter/view/f0$b$a;, "a;b;c;d;e;f", instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->a:F, instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->b:F, instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->c:F, instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->d:F, instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->e:F, instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->f:F)@Ljava/lang/runtime/ObjectMethods;->bootstrap(Ljava/lang/invoke/MethodHandles$Lookup;Ljava/lang/String;Ljava/lang/invoke/TypeDescriptor;Ljava/lang/Class;Ljava/lang/String;[Ljava/lang/invoke/MethodHandle;)Ljava/lang/Object;
+    if-ne p0, p1, :cond_0
 
     .line 2
+    const/4 p1, 0x1
+
     .line 3
+    return p1
+
     .line 4
-    move-result p0
+    :cond_0
+    instance-of v0, p1, Lcom/oplus/camera/feature/filter/view/f0$b$a;
 
     .line 5
-    return p0
+    const/4 v1, 0x0
+
+    .line 6
+    if-nez v0, :cond_1
+
+    .line 7
+    return v1
+
+    .line 8
+    :cond_1
+    check-cast p1, Lcom/oplus/camera/feature/filter/view/f0$b$a;
+
+    .line 9
+    iget v0, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->a:F
+
+    .line 10
+    iget v2, p1, Lcom/oplus/camera/feature/filter/view/f0$b$a;->a:F
+
+    .line 11
+    invoke-static {v0, v2}, Ljava/lang/Float;->compare(FF)I
+
+    .line 12
+    move-result v0
+
+    .line 13
+    if-eqz v0, :cond_2
+
+    .line 14
+    return v1
+
+    .line 15
+    :cond_2
+    iget v0, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->b:F
+
+    .line 16
+    iget v2, p1, Lcom/oplus/camera/feature/filter/view/f0$b$a;->b:F
+
+    .line 17
+    invoke-static {v0, v2}, Ljava/lang/Float;->compare(FF)I
+
+    .line 18
+    move-result v0
+
+    .line 19
+    if-eqz v0, :cond_3
+
+    .line 20
+    return v1
+
+    .line 21
+    :cond_3
+    iget v0, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->c:F
+
+    .line 22
+    iget v2, p1, Lcom/oplus/camera/feature/filter/view/f0$b$a;->c:F
+
+    .line 23
+    invoke-static {v0, v2}, Ljava/lang/Float;->compare(FF)I
+
+    .line 24
+    move-result v0
+
+    .line 25
+    if-eqz v0, :cond_4
+
+    .line 26
+    return v1
+
+    .line 27
+    :cond_4
+    iget v0, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->d:F
+
+    .line 28
+    iget v2, p1, Lcom/oplus/camera/feature/filter/view/f0$b$a;->d:F
+
+    .line 29
+    invoke-static {v0, v2}, Ljava/lang/Float;->compare(FF)I
+
+    .line 30
+    move-result v0
+
+    .line 31
+    if-eqz v0, :cond_5
+
+    .line 32
+    return v1
+
+    .line 33
+    :cond_5
+    iget v0, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->e:F
+
+    .line 34
+    iget v2, p1, Lcom/oplus/camera/feature/filter/view/f0$b$a;->e:F
+
+    .line 35
+    invoke-static {v0, v2}, Ljava/lang/Float;->compare(FF)I
+
+    .line 36
+    move-result v0
+
+    .line 37
+    if-eqz v0, :cond_6
+
+    .line 38
+    return v1
+
+    .line 39
+    :cond_6
+    iget v0, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->f:F
+
+    .line 40
+    iget p1, p1, Lcom/oplus/camera/feature/filter/view/f0$b$a;->f:F
+
+    .line 41
+    invoke-static {v0, p1}, Ljava/lang/Float;->compare(FF)I
+
+    .line 42
+    move-result p1
+
+    .line 43
+    if-nez p1, :cond_7
+
+    .line 44
+    const/4 v1, 0x1
+
+    .line 45
+    :cond_7
+    return v1
 .end method
 
 .method public final hashCode()I
-    .locals 0
+    .locals 3
 
     .line 1
-    invoke-custom {p0}, call_site_1("hashCode", (Lcom/oplus/camera/feature/filter/view/f0$b$a;)I, Lcom/oplus/camera/feature/filter/view/f0$b$a;, "a;b;c;d;e;f", instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->a:F, instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->b:F, instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->c:F, instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->d:F, instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->e:F, instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->f:F)@Ljava/lang/runtime/ObjectMethods;->bootstrap(Ljava/lang/invoke/MethodHandles$Lookup;Ljava/lang/String;Ljava/lang/invoke/TypeDescriptor;Ljava/lang/Class;Ljava/lang/String;[Ljava/lang/invoke/MethodHandle;)Ljava/lang/Object;
+    iget v0, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->a:F
 
     .line 2
+    invoke-static {v0}, Ljava/lang/Float;->floatToIntBits(F)I
+
     .line 3
+    move-result v0
+
     .line 4
-    move-result p0
+    mul-int/lit8 v0, v0, 0x1f
 
     .line 5
-    return p0
+    iget v1, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->b:F
+
+    .line 6
+    invoke-static {v1}, Ljava/lang/Float;->floatToIntBits(F)I
+
+    .line 7
+    move-result v1
+
+    .line 8
+    add-int/2addr v0, v1
+
+    .line 9
+    mul-int/lit8 v0, v0, 0x1f
+
+    .line 10
+    iget v1, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->c:F
+
+    .line 11
+    invoke-static {v1}, Ljava/lang/Float;->floatToIntBits(F)I
+
+    .line 12
+    move-result v1
+
+    .line 13
+    add-int/2addr v0, v1
+
+    .line 14
+    mul-int/lit8 v0, v0, 0x1f
+
+    .line 15
+    iget v1, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->d:F
+
+    .line 16
+    invoke-static {v1}, Ljava/lang/Float;->floatToIntBits(F)I
+
+    .line 17
+    move-result v1
+
+    .line 18
+    add-int/2addr v0, v1
+
+    .line 19
+    mul-int/lit8 v0, v0, 0x1f
+
+    .line 20
+    iget v1, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->e:F
+
+    .line 21
+    invoke-static {v1}, Ljava/lang/Float;->floatToIntBits(F)I
+
+    .line 22
+    move-result v1
+
+    .line 23
+    add-int/2addr v0, v1
+
+    .line 24
+    mul-int/lit8 v0, v0, 0x1f
+
+    .line 25
+    iget v1, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->f:F
+
+    .line 26
+    invoke-static {v1}, Ljava/lang/Float;->floatToIntBits(F)I
+
+    .line 27
+    move-result v1
+
+    .line 28
+    add-int/2addr v0, v1
+
+    .line 29
+    return v0
 .end method
 
 .method public final toString()Ljava/lang/String;
-    .locals 0
+    .locals 2
 
     .line 1
-    invoke-custom {p0}, call_site_2("toString", (Lcom/oplus/camera/feature/filter/view/f0$b$a;)Ljava/lang/String;, Lcom/oplus/camera/feature/filter/view/f0$b$a;, "a;b;c;d;e;f", instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->a:F, instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->b:F, instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->c:F, instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->d:F, instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->e:F, instance-get@Lcom/oplus/camera/feature/filter/view/f0$b$a;->f:F)@Ljava/lang/runtime/ObjectMethods;->bootstrap(Ljava/lang/invoke/MethodHandles$Lookup;Ljava/lang/String;Ljava/lang/invoke/TypeDescriptor;Ljava/lang/Class;Ljava/lang/String;[Ljava/lang/invoke/MethodHandle;)Ljava/lang/Object;
+    new-instance v0, Ljava/lang/StringBuilder;
 
     .line 2
+    invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
+
     .line 3
+    const-string v1, "a[a="
+
     .line 4
-    move-result-object p0
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
 
     .line 5
-    return-object p0
+    iget v1, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->a:F
+
+    .line 6
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(F)Ljava/lang/StringBuilder;
+
+    .line 7
+    const-string v1, ", b="
+
+    .line 8
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    .line 9
+    iget v1, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->b:F
+
+    .line 10
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(F)Ljava/lang/StringBuilder;
+
+    .line 11
+    const-string v1, ", c="
+
+    .line 12
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    .line 13
+    iget v1, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->c:F
+
+    .line 14
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(F)Ljava/lang/StringBuilder;
+
+    .line 15
+    const-string v1, ", d="
+
+    .line 16
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    .line 17
+    iget v1, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->d:F
+
+    .line 18
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(F)Ljava/lang/StringBuilder;
+
+    .line 19
+    const-string v1, ", e="
+
+    .line 20
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    .line 21
+    iget v1, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->e:F
+
+    .line 22
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(F)Ljava/lang/StringBuilder;
+
+    .line 23
+    const-string v1, ", f="
+
+    .line 24
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    .line 25
+    iget v1, p0, Lcom/oplus/camera/feature/filter/view/f0$b$a;->f:F
+
+    .line 26
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(F)Ljava/lang/StringBuilder;
+
+    .line 27
+    const-string v1, "]"
+
+    .line 28
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    .line 29
+    invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    .line 30
+    move-result-object v0
+
+    .line 31
+    return-object v0
 .end method
-- 
2.55.0

